File: void-lib/void-separate-modules/void-sql-builder/src/main/kotlin/cn/toutatis/xvoid/sql/base/XFunc.java
package cn.toutatis.xvoid.sql.base;

import java.io.Serializable;
import java.util.function.Function;

/**
 * @author Toutatis_Gc
 */
@FunctionalInterface
public interface XFunc<T,R> extends Function<T,R>, Serializable {
}

File: void-lib/void-separate-modules/void-toolkit/src/main/kotlin/cn/toutatis/xvoid/toolkit/clazz/XFunc.kt
package cn.toutatis.xvoid.toolkit.clazz

import java.io.Serializable
import java.lang.FunctionalInterface
import java.util.function.Function

/**
 * @see LambdaToolkit 供LambdaToolkit使用
 * @author Toutatis_Gc
 */
@FunctionalInterface
interface XFunc<T, R> : Function<T, R>, Serializable